### ISSUE TITLE
Call.absoluteURL and HTTPS

### ISSUE THREAD

**Issue by guillaumeprevot** [playframework/playframework#2427]
This is the first time I contact you so it's time to say "Thank you !" for this wonderfull plateform. It's so easy and efficient..

When working with HTTPS, play.mvc.Call.absoluteURL(...) may not work properly because "secure" parameter may be inconsistent with request.host(), which is host:port.

Example : request on https://localhost:9001
- wrong : absoluteURL(request) will generate http://localhost:9001
- wrong : absoluteURL(request, false) will generate http://localhost:9001
- ok : absoluteURL(request, true) will generate https://localhost:9001

Example : request on http://localhost:9000
- ok : absoluteURL(request) will generate http://localhost:9000
- ok : absoluteURL(request, false) will generate http://localhost:9000
- wrong : absoluteURL(request, true) will generate https://localhost:9000

Maybe you should force port in URL depending on "secure" parameter.

PS: same issue with webSocketURL

**Comment by nmrais** [comment]
I am having issue redirecting to an external URL from play controller. The base URI is missing and it only contains the relative URL. It works when accessing from a play application, but when this redirect is consumed by an external application (non play), they only receive the relative path. Any idea how we can get the absolute external url.??
Suggestions please.

**Comment by guillaumeprevot** [comment]
I think the answer from jroper is still valid. There's no way to known on server side the absolute URL used on client side to access page. As you said, the communication may involve third party applications (like proxies, ...). That's why I ended up using a settings in applicaton.conf for the base URL (which you would know as soon as you deploy your application).
Maybe you could try this :
- add a custom settings in application.conf for the base URL to use
- concatenate this base URL with the relative URL to get an absolute URL
- send a redirect with the absolute URL

**Comment by nmrais** [comment]
Thank you so much for your reply. The issue I just found was that the URL which I generated was missing a session ID which gets created when we post the URL on the web browser. So, I am trying to figure out a way to post this URL and get the response and get the updated URL from the response and then do the redirect. I will explain below

The complete URL I generate is : https://xxx.ttt.com?uniqueId=1234 and when I paste it on the browser and submit, I get https://xxx.ttt.com/DWEDSDFDFFFDFDWE/$/?uniqueId=1234. Now, when I hard code this URL and return from the controller, it redirects successfully, but the value DWEDSDFDFFFDFDWE/$ expires in 5 minutes, so I was thinking of a way to invoke the URL and get the response and get this sessionID or whatever code and append to the actual URL and then do a redirect. Any helps on that?
